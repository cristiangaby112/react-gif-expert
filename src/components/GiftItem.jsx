import PropTypes from "prop-types";


const GiftItem = ({ title, url }) => {
  // console.log(title)
  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{title}</p>
    </div>
  )
}
GiftItem.propTypes ={
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
export default GiftItem;
