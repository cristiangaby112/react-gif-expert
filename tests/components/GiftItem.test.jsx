import { render, screen } from "@testing-library/react"
import GiftItem from "../../src/components/GiftItem"






describe('Prueba de GifItem', () =>{

    const title ='Dragon ball';
    const url = 'https://dragon-ball/goku.jpg';


    test('Debe hacer match con el snapshot', () => { 

        const {container} = render(<GiftItem title={title} url={url}/>)
        // console.log(container)
        expect(container).toMatchSnapshot();


     })

     test('Debe de mostrar la imagen con el URL Y el ALT indicado', () => { 

        render(<GiftItem title={title} url={url}/>)
        // screen.debug();
        // console.log(screen.getByRole('img').src)
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt} = screen.getByRole('img');
        expect( src ).toBe(url)
        expect( alt ).toBe(title)
      });

      test('debe de mostar el title en el componente', () => { 
        
        render(<GiftItem title={title} url={url}/>)
        expect(title).toBeTruthy()

       })
})