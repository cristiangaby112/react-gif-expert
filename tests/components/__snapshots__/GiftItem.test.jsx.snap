// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://dragon-ball/goku.jpg"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
